### DISCUSSION TITLE
more flexibility with materials in time series files

### DISCUSSION THREAD

**Discussion by shabbinga** [tkeskita/BVtkNodes#81] [Ideas]
Hi, 
I am currently working with time-dependent vtu and vtp data, i.e. fileseries. I stumbled over the fact that with the setting "generate material = true" you lose all settings like for example the alpha value, which you had set afterwards on the material, as soon as you select a new time step.
In the case of the alpha value a rather simple solution is possible: 
In the ColorRamp in the BVtk nodetree you can also specify an alpha value for the individual colors. However, this would have to be connected to the Principled BSDF in the Material nodes.This works, if I add in the file converters.py in the "create_material" function the line 
 links.new(node3.outputs["Alpha"], node2.inputs["Alpha"])

Maybe this could be added to the next version of the BVtkNodes-Addon? Other values, such as the roughness and the specular value, would also be great to be able to adjust in BVtkNodes. 
Best regards, 
Sonja Habbinga

**Comment by tkeskita** [comment]
Hi,

connecting those alpha's is easy, I'll add that. But in general I'd suggest disabling Generate Material after finetuning the material in Blender to avoid it getting overwritten.

It would be nice to be able to use something else than just the four basic color channels (RGBA) to get information from VTK to Material Nodes. Maybe Vertex Color? I haven't tried that though..
